src/app/client/src/app/modules/groups/components/group-form/group-form.component.ts

```diff
@@ -1,36 +1,251 @@
+    this.userService.userData$.subscribe(userdata => {
+      if (userdata && !userdata.err) {
+        this.userProfile = userdata.userProfile;
+      }
+    });
+    this.selectedOption = _.pickBy(_.cloneDeep(this.userProfile.framework), 'length') || {};
+    this.groupForm = this.fb.group({
+      groupName: ['', [ Validators.required ]],
+      board: [_.get(this.selectedOption, 'board') || [], [ Validators.required ]],
+      medium: [_.get(this.selectedOption, 'medium') || [], [ Validators.required ]],
+      gradeLevel: [_.get(this.selectedOption, 'gradeLevel') || [], [ Validators.required ]],
+      subject: [_.get(this.selectedOption, 'subject') || [], [ Validators.required ]],
+    });
+    this.editMode = _.some(this.selectedOption, 'length') || false ;
+    this.unsubscribe = this.isCustodianOrgUser().pipe(
+      mergeMap((custodianOrgUser: boolean) => {
+        this.custodianOrg = custodianOrgUser;
+        if (custodianOrgUser) {
+          return this.getFormOptionsForCustodianOrg();
+        } else {
+          return this.getFormOptionsForOnboardedUser();
+        }
+      }), first()).subscribe(data => {
+        this.formFieldOptions = data;
+        this.boardList = _.get(_.find(data, { 'code': 'board' }), 'range') || [];
+        this.mediumList = _.get(_.find(data, { 'code': 'medium' }), 'range') || [];
+        this.gradesList = _.get(_.find(data, { 'code': 'gradeLevel' }), 'range') || [];
+        this.subjectList = _.get(_.find(data, { 'code': 'subject' }), 'range') || [];
+      }, err => {
+        this.toasterService.warning(this.resourceService.messages.emsg.m0012);
+      });
+  }
+  private isCustodianOrgUser() {
+    return this.orgDetailsService.getCustodianOrg().pipe(map((custodianOrg) => {
+      if (_.get(this.userService, 'userProfile.rootOrg.rootOrgId') === _.get(custodianOrg, 'result.response.value')) {
+        return true;
+      }
+      return false;
+    }));
+  }
+  private getFormOptionsForCustodianOrg() {
+    return this.getCustodianOrgData().pipe(mergeMap((data) => {
+      this.custodianOrgBoard = data;
+      const boardObj = _.cloneDeep(this.custodianOrgBoard);
+      boardObj.range = _.sortBy(boardObj.range, 'index');
+      const board = boardObj;
+      if (_.get(this.selectedOption, 'board[0]')) { // update mode, get 1st board framework and update all fields
+        this.selectedOption.board = _.get(this.selectedOption, 'board[0]');
+        this.frameWorkId = _.get(_.find(this.custOrgFrameworks, { 'name': this.selectedOption.board }), 'identifier');
+        return this.getFormatedFilterDetails().pipe(map((formFieldProperties) => {
+          this._formFieldProperties = formFieldProperties;
+          this.mergeBoard(); // will merge board from custodian org and board from selected framework data
+          return this.getUpdatedFilters(board, true);
+        }));
+      } else {
+        return of([ board ]);
+      }
+    }));
+  }
+  private getFormOptionsForOnboardedUser() {
```

Comment:
Move the code(methods) which is not interacting with components template to service so that it is lean

Reply:
Resolve